Sources/MongoKitten/Aggregate.swift

```diff
@@ -104,6 +104,12 @@ public struct AggregateBuilderPipeline: QueryCursor {
         let command = makeCommand()
         let connection = try await getConnection()
         
+        let minimalVersionRequired = stages.compactMap { $0.minimalVersionRequired }.min()
```

Comment:
I assume that you need the `maximum` version in that list of stages, right? So if one stage requires MongoDB 3.4, and another requires 3.6, you'll want to check if the server is 3.6 or greater.

Reply:
Also, this should be changed into a keypath accessor:
```swift
stages.compactMap(\.minimalVersionRequired)
```